proto/viam/app/datasync/v1/data_sync.proto

```diff
@@ -10,9 +10,26 @@ import "tagger/v1/tagger.proto";
 option go_package = "go.viam.com/api/app/datasync/v1";
 
 service DataSyncService {
-  rpc Upload(stream viam.app.datasync.v1.UploadRequest) returns (stream viam.app.datasync.v1.UploadResponse);
+  rpc DataCaptureUpload(viam.app.datasync.v1.DataCaptureUploadRequest) returns (viam.app.datasync.v1.DataCaptureUploadResponse);
+  rpc FileUpload(stream viam.app.datasync.v1.FileUploadRequest) returns (viam.app.datasync.v1.FileUploadResponse);
 }
 
+message DataCaptureUploadRequest {
+  UploadMetadata metadata = 1;
+  repeated SensorData sensor_contents = 2;
+}
+
+message DataCaptureUploadResponse {}
+
+message FileUploadRequest {
+  oneof upload_packet {
```

Comment:
@edaniels, are we supposed to be avoiding `oneof` in requests as well as responses?

Reply:
I think if oneof can be avoided easily, it should. In this case it seems okay to have a stream use a oneof since it represents stages. It needs comments though as to how to use it properly.

Reply:
+1, and we'll reuse these comments in the PackageService's CreatePackageRequest oneof.